crates/fuel-core/src/service/sub_services.rs

```diff
@@ -59,30 +59,37 @@ pub fn init_sub_services(
     config: &Config,
     database: CombinedDatabase,
 ) -> anyhow::Result<(SubServices, SharedState)> {
+    let chain_config = config.snapshot_reader.chain_config();
```

Comment:
I know that this isn't your change, but it feels bad that our `chain_config` is inside our `config.snapshot_reader`. Like, that's very unintuitive for a couple reasons: 

1. what's the difference between a `Config` and a `ChainConfig`?
1. why would anyone looking for the `consensus` (or any other `ChainConfig` value) go to the `snapshot_reader` to find that? Like, even now that I've found it, I'm worried that there must be another field by the same name someplace else because this feels namespaced into some snapshot sub-category.